udaykalra/walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -13,11 +13,133 @@
 // limitations under the License.
 
 package com.google.sps;
-
+import java.util.ArrayList;
 import java.util.Collection;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
 
+/**
+ * Returns a collection of TimeRange objects to indicate available meeting times
+ * for indicated people and duration given through a request object.
+ */
 public final class FindMeetingQuery {
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+
+    /* Create an output collection with a full day to cut(shorten or split) 
+     * into the available ranges.
+     */
+    TimeRange fullDay = TimeRange.WHOLE_DAY;
+    Collection<TimeRange> availableRanges = new HashSet<>();
+    availableRanges.add(fullDay);
+
+    // Obtain data from the request.
+    Collection<String> attendees = request.getAttendees();
+    long requestDuration = request.getDuration();
+
+    Collection<TimeRange> busyTimes = new HashSet<>();
```

Comment:
It would help if naming of collections holding TimeRange objects were consistent throughout. This is named `busyTimes`, but the above collection is `availableRanges`. Could both be named `Ranges` or `Times`? This reduces cognitive load while reading.

Reply:
Done